### PR TITLE
[Draft] Introduce textlint default terminology rules

### PR DESCRIPTION
**Pull request by svrnm**
@chalin these are the changes the default rule set would introduce (as discussed in #3001). Overall they look good to me, I fixed the one end-user case we discussed recently.

A compressed version of the suggested changes:

```
Incorrect usage of the term: “3rd-party”, use “third-party” instead
Incorrect usage of the term: “AJAX”, use “Ajax” instead
Incorrect usage of the term: “Bug fixes”, use “Bugfixes”
Incorrect usage of the term: “Bug fixes”, use “bugfixes”
Incorrect usage of the term: “CLI tool”, use “command-line
Incorrect usage of the term: “Co-located”, use “Colocated” instead
Incorrect usage of the term: “Co-located”, use “colocated” instead
Incorrect usage of the term: “Command line”, use “Command-line”
Incorrect usage of the term: “End-User”, use “End user”
Incorrect usage of the term: “End-User”, use “end user”
Incorrect usage of the term: “End-user”, use “End user”
Incorrect usage of the term: “Front-end”, use “Frontend” instead
Incorrect usage of the term: “Github”, use “GitHub” instead
Incorrect usage of the term: “ID's”, use “IDs” instead
Incorrect usage of the term: “Ids”, use “IDs” instead
Incorrect usage of the term: “Id”, use “ID” instead
Incorrect usage of the term: “Javascript”, use “JavaScript” instead
Incorrect usage of the term: “Key/value”, use “Key-value” instead
Incorrect usage of the term: “MacOS”, use “macOS” instead
Incorrect usage of the term: “Macbook”, use “MacBook” instead
Incorrect usage of the term: “NPM”, use “npm” instead
Incorrect usage of the term: “Ok”, use “OK” instead
Incorrect usage of the term: “Repo”, use “repository” instead
Incorrect usage of the term: “Typescript”, use “TypeScript” instead
Incorrect usage of the term: “Website”, use “Site” instead
Incorrect usage of the term: “Wordpress”, use “WordPress” instead
Incorrect usage of the term: “api”, use “API” instead
Incorrect usage of the term: “back-ends”, use “backends” instead
Incorrect usage of the term: “back-end”, use “backend” instead
Incorrect usage of the term: “backwards compatible”, use “backward
Incorrect usage of the term: “bug fixes”, use “bugfixes”
Incorrect usage of the term: “bug fix”, use “bugfix”
Incorrect usage of the term: “built in”, use “built-in”
Incorrect usage of the term: “cli tool”, use “command-line
Incorrect usage of the term: “client side”, use “client-side”
Incorrect usage of the term: “co-located”, use “colocated” instead
Incorrect usage of the term: “command line”, use “command-line”
Incorrect usage of the term: “end-points”, use “endpoints” instead
Incorrect usage of the term: “end-users”, use “end users”
Incorrect usage of the term: “end-user”, use “end user”
Incorrect usage of the term: “front-end”, use “frontend” instead
Incorrect usage of the term: “github”, use “GitHub” instead
Incorrect usage of the term: “host names”, use “hostnames”
Incorrect usage of the term: “html”, use “HTML” instead
Incorrect usage of the term: “ids”, use “IDs” instead
Incorrect usage of the term: “id”, use “ID” instead
Incorrect usage of the term: “key/value”, use “key-value” instead
Incorrect usage of the term: “meta-data”, use “metadata” instead
Incorrect usage of the term: “ok”, use “OK” instead
Incorrect usage of the term: “pre-defined”, use “predefined” instead
Incorrect usage of the term: “regex”, use “regular expression”
Incorrect usage of the term: “repo”, use “repository” instead
Incorrect usage of the term: “run-time”, use “runtime” instead
Incorrect usage of the term: “screen shot”, use “screenshot”
Incorrect usage of the term: “server side”, use “server-side”
Incorrect usage of the term: “time stamp”, use “timestamp”
Incorrect usage of the term: “typescript”, use “TypeScript” instead
Incorrect usage of the term: “url”, use “URL” instead
Incorrect usage of the term: “web sites”, use “sites”
Incorrect usage of the term: “web site”, use “site”
Incorrect usage of the term: “websites”, use “sites” instead
Incorrect usage of the term: “website”, use “site” instead
```

### CODE REVIEW

File: content/en/blog/2022/instrument-kafka-clients/index.md [hunk 1]
```diff
@@ -163,7 +163,7 @@ The main environment variables to be set are the following:
 
 Instead of using the above environment variables, it is also possible to use
 corresponding system properties to be set programmatically in the code or on the
-command line when launching the application They are `otel.service.name` and
+command-line when launching the application They are `otel.service.name` and
```
Comment (chalin) [inline content/en/blog/2022/instrument-kafka-clients/index.md:166]:
```suggestion
command line when launching the application They are `otel.service.name` and
```

File: content/en/blog/2022/new-end-user-resources.md [hunk 2]
```diff
@@ -10,5 +10,5 @@ date: 2022-06-30
 ---
 
 The content of this post has moved to
-[End-user Resources](/community/end-user/), where it will be kept up-to-date as
-more end-user resources become available.
+[End user Resources](/community/end-user/), where it will be kept up-to-date as
+more end user resources become available.
```
Comment (chalin) [inline content/en/blog/2022/new-end-user-resources.md:14]: Should be hyphenated here.

**Comment by chalin** [comment]
Are the defaults an all-or-thing deal? Or can we turn it on, and disable a few of the rules? E.g., I'd prefer that we continue to use:

- "website" vs. "site"
- Whether "end-user" is hyphenated or not, depends on the context (whether "end user" is an adjective).
- The same remark applies to "command line"

**Review by chalin — COMMENTED** [review]
Some initial feedback; see inline comments.

**Comment by svrnm** [comment]
> Are the defaults an all-or-thing deal? Or can we turn it on, and disable a few of the rules? 

From what I understand is that you can turn them on and then disable a few you don't like. I agree with the ones you picked, so let me see how I can disable them

**Comment by chalin** [comment]
That works for me too. The original goal was to fix "ID".

**Pull request closed without merging**